File: src/main/java/com/characohol/characohol_api/repository/RecommendationRepository.java
package com.characohol.characohol_api.repository;

import com.characohol.characohol_api.domain.Recommendation;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface RecommendationRepository extends JpaRepository<Recommendation, Long> {
    Optional<Recommendation> findByTag(String tag);
}

File: src/main/kotlin/com/characohol/characohol_api/repository/RecommendationRepository.kt
package com.characohol.characohol_api.repository

import com.characohol.characohol_api.domain.Recommendation
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Mono

interface RecommendationRepository : ReactiveCrudRepository<Recommendation, Long> {
    fun findByTag(tag: String): Mono<Recommendation>
}